security/pkg/server/ca/server.go

```diff
@@ -182,37 +183,51 @@ func (s *Server) HandleCSR(ctx context.Context, request *pb.CsrRequest) (*pb.Csr
 
 // Run starts a GRPC server on the specified port.
 func (s *Server) Run() error {
-	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", s.port))
-	if err != nil {
-		return fmt.Errorf("cannot listen on port %d (error: %v)", s.port, err)
-	}
+	grpcServer := s.grpcServer
+	var listener net.Listener
+	var err error
+
+	if grpcServer == nil {
+		listener, err = net.Listen("tcp", fmt.Sprintf(":%d", s.port))
+		if err != nil {
+			return fmt.Errorf("cannot listen on port %d (error: %v)", s.port, err)
+		}
 
-	var grpcOptions []grpc.ServerOption
-	grpcOptions = append(grpcOptions, s.createTLSServerOption(), grpc.UnaryInterceptor(grpc_prometheus.UnaryServerInterceptor))
+		var grpcOptions []grpc.ServerOption
+		grpcOptions = append(grpcOptions, s.createTLSServerOption(), grpc.UnaryInterceptor(grpc_prometheus.UnaryServerInterceptor))
 
-	grpcServer := grpc.NewServer(grpcOptions...)
+		grpcServer = grpc.NewServer(grpcOptions...)
+	}
 	pb.RegisterIstioCAServiceServer(grpcServer, s)
 	pb.RegisterIstioCertificateServiceServer(grpcServer, s)
 
 	grpc_prometheus.EnableHandlingTimeHistogram()
 	grpc_prometheus.Register(grpcServer)
 
-	// grpcServer.Serve() is a blocking call, so run it in a goroutine.
-	go func() {
-		serverCaLog.Infof("Starting GRPC server on port %d", s.port)
+	if listener != nil {
+		// grpcServer.Serve() is a blocking call, so run it in a goroutine.
+		go func() {
+			serverCaLog.Infof("Starting GRPC server on port %d", s.port)
 
-		err := grpcServer.Serve(listener)
+			err := grpcServer.Serve(listener)
```

Comment:
Restore the indent here?

Reply:
yes, make format-go will take care of all this. Cut&paste...